File: front/app/features/ToDoTasks/dto/Task.ts
import type { Category } from "./Category";
import type { Priority } from "./Priority";

export interface Task {
  id: number;
  title: string;
  cycles: number;
  cyclesCurrent: number;
  status: "in progress" | "to do" | "done" | "canceled";
  priority: Priority | null;
  category: Category | null;
}

File: user-service/src/main/java/com/example/user_service/Entity/Task.java
package com.example.user_service.Entity;

import lombok.*;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Task {
    private Long id;

    private String title;

    private String status;

    private int cycles;

    private int cyclesCurrent;

    private Priority priority;

    private Category category;

    private String userId;
}
